[New Entry/ Edits] Python Builtin-Functions: `super()`/ Inheritance

beverand opened the pull request:
<!--
👋 Hi, thanks for sending a PR to Codecademy Docs! 💖
Please fill out all fields below to ensure your PR is reviewed quickly.

**IMPORTANT**

If you would like to receive credit for your contribution to an entry, make sure to have your Codecademy user profile linked to your GitHub account:

1. Go to your Codecademy dashboard.
2. Click on your profile image in the top-right and then choose "Profile".
3. Then click "Edit Profile".
4. In the "GitHub Username" field, add your username (without the @).
5. Click "Save Changes"

Of course, you can opt not to do this and be listed as an "Anonymous contributor", instead. :)
-->

### Description

<!-- Please write a summary of the change, such as which topic(s) and file(s) that you have edited or created. Please also include relevant motivation and context: -->

Closes #586

### Type of Change

<!--- Please delete or cross off the bullet point(s) that are irrelevant to this PR: -->

- Adding a new entry
- Editing an existing entry (fixing a typo, bug, issues, etc)

### Checklist

<!-- Please check ALL the boxes: -->

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

<!--
Having trouble with the PR checker? Here are some common issues and resolutions:

- verify_formatting is failing
  - run `yarn format path/to/markdown/file.md` or `yarn format:all` and commit the results
- verify_lint is failing
  - same as above
  - if verify_lint is still failing, running `yarn lint` locally should let you know what needs to be changed by hand
- test is failing
  - ensure any new markdown files have a `Title` and `Description` defined in their metadata
  - ensure any new markdown files only contain alphanumerics and dashes in their file names and have the same name as their parent directory
  - if that looks ok, running `yarn test` locally should let you know what the issue is
-->

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/Codecademy/docs?pullRequest=1008) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/Codecademy/docs?pullRequest=1008) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/Codecademy/docs?pullRequest=1008) it.</sub>

beverand commented:
Added the super() term as discussed in issue #586
- created super.md under python/concepts/built-in-functions/terms/super
- removed super() from python inheritance entry

Dusch4593 commented:
> Added the super() term as discussed in issue #586
> 
> * created super.md under python/concepts/built-in-functions/terms/super
> * removed super() from python inheritance entry

@beverand Thanks for submitting this PR! I'm looking forward to conducting the first review for this 😄

Quick tip: For your next PR, if you write "Closes #[this issue number]" in the "Description" near the top of the PR, the related issue will be linked to the PR. (No worries for this one, though; just edited the description 😎)

Dusch4593 commented on content/python/concepts/built-in-functions/terms/super/super.md:
```diff
@@ -0,0 +1,53 @@
+Title: 'super()'
+Description: 'implicitly refer to the superclass'
+Subjects:
+  - 'Computer Science'
```
```suggestion
  - 'Computer Science'
  - 'Data Science'
```

Since Python is commonly used in data science.

beverand replied:
I've added Data Science as a tag. I've actually never encountered super() as a data science term that why I didn't put it in, but I admit, I haven't taken Codecademy's Data Science course.

Dusch4593 replied:
Yeah these are relatively loose associations meant to bolster the entry's SEO.

Dusch4593 commented on content/python/concepts/built-in-functions/terms/super/super.md:
```diff
@@ -0,0 +1,53 @@
+  - 'paths/computer-science'
+---
+
+The `super()` function is built into python and is used to refer to the the superclass of an object without specifically using the parent name.
```
```suggestion
The **`super()`** function returns a temporary object that allows a given [class](https://www.codecademy.com/resources/docs/python/classes) to [inherit](https://www.codecademy.com/resources/docs/python/inheritance) the methods and properties of a parent or sibling class.
```

beverand replied:
I've made this modification.

Dusch4593 commented on content/python/concepts/built-in-functions/terms/super/super.md:
```diff
@@ -0,0 +1,53 @@
+
+`super()` returns a temporary object of the superclass, allowing the superclass’s methods to be called. This comes in handy when using [multiple inheritance](#multiple-inheritance).
+
+## Example
```
```suggestion
## Codebyte Example

```

Let's allow one empty line above and below each header.

Dusch4593 replied:
Also let's rename this to "Codebyte Example" since Python is a supported language for [runnable code snippets](https://help.codecademy.com/hc/en-us/articles/1260806285369-Codebytes) (which is compatible with Codecademy Docs 😄)

beverand replied:
Updated

Dusch4593 commented on content/python/concepts/built-in-functions/terms/super/super.md:
````diff
@@ -0,0 +1,53 @@
+## Example
+In this case, the `.say_hi()` method is implemented utilizing `super()` inside `.intro()`:
+
+```py
````
```suggestion
```codebyte/python
```

beverand replied:
Updated!

Dusch4593 commented on content/python/concepts/built-in-functions/terms/super/super.md:
````diff
@@ -0,0 +1,53 @@
+doc = Python()
+doc.intro()
+```
+
+The output would be:
+
+```shell
+Hi! I am a Programming Language.
+Python here!
+```
````
```suggestion
```

If we choose to go with a Codebyte example, then these lines can be omitted; the output will appear on the screen after the code is run.

beverand replied:
I've removed this portion of the doc.

Dusch4593 commented on content/python/concepts/inheritance/inheritance.md:
```diff
@@ -68,28 +68,6 @@ Python here!
 
 **Note**: Inside the child class, the same number and type of arguments need to be passed into the parent class' method.
```
```suggestion
> **Note:** Inside the child class, the same number and type of arguments need to be passed into the parent class' method.
```

Dusch4593 replied:
Also, near the top on line # 16, can we **bold** the word "Inheritance"?

beverand replied:
Updated.

Dusch4593 commented on content/python/concepts/built-in-functions/terms/super/super.md:
```diff
@@ -0,0 +1,53 @@
+Subjects:
+  - 'Computer Science'
+Tags:
+  - 'Functions'
+  - 'Objects'
+  - 'Inheritance'
```
```suggestion
  - 'Classes'
  - 'Functions'
  - 'Inheritance'
  - 'Methods'
  - 'Objects'
  - 'OOP'
```

beverand replied:
Absolutely! Added.

Dusch4593 commented on content/python/concepts/built-in-functions/terms/super/super.md:
```diff
@@ -0,0 +1,53 @@
+---
+Title: 'super()'
+Description: 'implicitly refer to the superclass'
```
```suggestion
Description: 'Returns a temporary object that allows a given class to inherit the methods and properties of a parent or sibling class.'
```

Let's make sure to use capitalization and punctuation. 😄

beverand replied:
Updated

Dusch4593 reviewed (changes requested):
@beverand Finished with my review and just left some comments/suggestions for you to check out. I might have more later but I wanted to give you a chance to look at what I have so far. Let me know if you have any questions. 😄

beverand commented:
Thank you so much for the review. I didn't think it was going to be perfect
so I was fully anticipating that corrections would need to be made. I will
try to get these modifications in tonight and resubmit.


On Fri, Sep 16, 2022 at 2:27 PM Brandon Dusch ***@***.***>
wrote:

> ***@***.**** requested changes on this pull request.
>
> @beverand <https://github.com/beverand> Finished with my review and just
> left some comments/suggestions for you to check out. I might have more
> later but I wanted to give you a chance to look at what I have so far. Let
> me know if you have any questions. 😄
> ------------------------------
>
> In content/python/concepts/built-in-functions/terms/super/super.md
> <https://github.com/Codecademy/docs/pull/1008#discussion_r973391079>:
>
> > @@ -0,0 +1,53 @@
>
> +---
>
> +Title: 'super()'
>
> +Description: 'implicitly refer to the superclass'
>
> +Subjects:
>
> +  - 'Computer Science'
>
>
> ⬇️ Suggested change
>
> -  - 'Computer Science'
>
> +  - 'Computer Science'
>
> +  - 'Data Science'
>
>
> Since Python is commonly used in data science.
> ------------------------------
>
> In content/python/concepts/built-in-functions/terms/super/super.md
> <https://github.com/Codecademy/docs/pull/1008#discussion_r973394614>:
>
> > @@ -0,0 +1,53 @@
>
> +---
>
> +Title: 'super()'
>
> +Description: 'implicitly refer to the superclass'
>
> +Subjects:
>
> +  - 'Computer Science'
>
> +Tags:
>
> +  - 'Functions'
>
> +  - 'Objects'
>
> +  - 'Inheritance'
>
> +CatalogContent:
>
> +  - 'learn-python-3'
>
> +  - 'paths/computer-science'
>
> +---
>
> +
>
> +The `super()` function is built into python and is used to refer to the the superclass of an object without specifically using the parent name.
>
>
> ⬇️ Suggested change
>
> -The `super()` function is built into python and is used to refer to the the superclass of an object without specifically using the parent name.
>
> +The **`super()`** function returns a temporary object that allows a given [class](https://www.codecademy.com/resources/docs/python/classes) to [inherit](https://www.codecademy.com/resources/docs/python/inheritance) the methods and properties of a parent or sibling class.
>
>
> ------------------------------
>
> In content/python/concepts/built-in-functions/terms/super/super.md
> <https://github.com/Codecademy/docs/pull/1008#discussion_r973397461>:
>
> > +---
>
> +
>
> +The `super()` function is built into python and is used to refer to the the superclass of an object without specifically using the parent name.
>
> +
>
> +## Syntax
>
> +
>
> +```pseudo
>
> +super().function()
>
> +super(type)
>
> +```
>
> +
>
> +## `super()`
>
> +
>
> +`super()` returns a temporary object of the superclass, allowing the superclass’s methods to be called. This comes in handy when using [multiple inheritance](#multiple-inheritance).
>
> +
>
> +## Example
>
>
> ⬇️ Suggested change
>
> -## Example
>
> +## Codebyte Example
>
>
> Let's allow one empty line above and below each header.
> ------------------------------
>
> In content/python/concepts/built-in-functions/terms/super/super.md
> <https://github.com/Codecademy/docs/pull/1008#discussion_r973399307>:
>
> > +
>
> +## Syntax
>
> +
>
> +```pseudo
>
> +super().function()
>
> +super(type)
>
> +```
>
> +
>
> +## `super()`
>
> +
>
> +`super()` returns a temporary object of the superclass, allowing the superclass’s methods to be called. This comes in handy when using [multiple inheritance](#multiple-inheritance).
>
> +
>
> +## Example
>
> +In this case, the `.say_hi()` method is implemented utilizing `super()` inside `.intro()`:
>
> +
>
> +```py
>
>
> ⬇️ Suggested change
>
> -```py
>
> +```codebyte/python
>
>
> ------------------------------
>
> In content/python/concepts/built-in-functions/terms/super/super.md
> <https://github.com/Codecademy/docs/pull/1008#discussion_r973399720>:
>
> > +
>
> +The output would be:
>
> +
>
> +```shell
>
> +Hi! I am a Programming Language.
>
> +Python here!
>
> +```
>
>
> ⬇️ Suggested change
>
> -
>
> -The output would be:
>
> -
>
> -```shell
>
> -Hi! I am a Programming Language.
>
> -Python here!
>
> -```
>
>
> If we choose to go with a Codebyte example, then these lines can be
> omitted; the output will appear on the screen after the code is run.
> ------------------------------
>
> In content/python/concepts/inheritance/inheritance.md
> <https://github.com/Codecademy/docs/pull/1008#discussion_r973400620>:
>
> > @@ -68,28 +68,6 @@ Python here!
>
>
>
>  **Note**: Inside the child class, the same number and type of arguments need to be passed into the parent class' method.
>
>
> ⬇️ Suggested change
>
> -**Note**: Inside the child class, the same number and type of arguments need to be passed into the parent class' method.
>
> +> **Note:** Inside the child class, the same number and type of arguments need to be passed into the parent class' method.
>
>
> ------------------------------
>
> In content/python/concepts/built-in-functions/terms/super/super.md
> <https://github.com/Codecademy/docs/pull/1008#discussion_r973401467>:
>
> > +  - 'Functions'
>
> +  - 'Objects'
>
> +  - 'Inheritance'
>
>
> ⬇️ Suggested change
>
> -  - 'Functions'
>
> -  - 'Objects'
>
> -  - 'Inheritance'
>
> +  - 'Classes'
>
> +  - 'Functions'
>
> +  - 'Inheritance'
>
> +  - 'Methods'
>
> +  - 'Objects'
>
> +  - 'OOP'
>
>
> ------------------------------
>
> In content/python/concepts/built-in-functions/terms/super/super.md
> <https://github.com/Codecademy/docs/pull/1008#discussion_r973401977>:
>
> > @@ -0,0 +1,53 @@
>
> +---
>
> +Title: 'super()'
>
> +Description: 'implicitly refer to the superclass'
>
>
> ⬇️ Suggested change
>
> -Description: 'implicitly refer to the superclass'
>
> +Description: 'Returns a temporary object that allows a given class to inherit the methods and properties of a parent or sibling class.'
>
>
> Let's make sure to use capitalization and punctuation. 😄
> ------------------------------
>
> In content/python/concepts/built-in-functions/terms/super/super.md
> <https://github.com/Codecademy/docs/pull/1008#discussion_r973399213>:
>
> > +---
>
> +
>
> +The `super()` function is built into python and is used to refer to the the superclass of an object without specifically using the parent name.
>
> +
>
> +## Syntax
>
> +
>
> +```pseudo
>
> +super().function()
>
> +super(type)
>
> +```
>
> +
>
> +## `super()`
>
> +
>
> +`super()` returns a temporary object of the superclass, allowing the superclass’s methods to be called. This comes in handy when using [multiple inheritance](#multiple-inheritance).
>
> +
>
> +## Example
>
>
> Also let's rename this to "Codebyte Example" since Python is a supported
> language for runnable code snippets
> <https://help.codecademy.com/hc/en-us/articles/1260806285369-Codebytes>
> (which is compatible with Codecademy Docs 😄)
> ------------------------------
>
> In content/python/concepts/inheritance/inheritance.md
> <https://github.com/Codecademy/docs/pull/1008#discussion_r973400845>:
>
> > @@ -68,28 +68,6 @@ Python here!
>
>
>
>  **Note**: Inside the child class, the same number and type of arguments need to be passed into the parent class' method.
>
>
> Also, near the top on line # 16, can we *bold* the word "Inheritance"?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Codecademy/docs/pull/1008#pullrequestreview-1111220391>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEZUG6WHXZNSDMRI3GVT4E3V6TQ5VANCNFSM6AAAAAAQOCNCAQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

Dusch4593 commented on content/python/concepts/inheritance/inheritance.md:
```diff
@@ -68,28 +68,6 @@ Python here!
 
 **Note**: Inside the child class, the same number and type of arguments need to be passed into the parent class' method.
 
```
```suggestion
```

Let's also delete this new line (which is probably causing the "lint" check to not pass).

beverand replied:
Got rid of it. Thanks.

Dusch4593 replied:
Hmm did we remove it? The GitHub checks still seems to not be passing. 🤔

beverand replied:
Done.

Dusch4593 commented:
@beverand I would undo the changes in 4e8541efd800092196ecd3ce0448fe31d8a83597. When I [suggested](https://github.com/Codecademy/docs/pull/1008/files#r973400845) bolding "Inheritance" I meant the edit should happen in line # 16 in the `inheritance.md` file (not the `super.md` file).

beverand commented:
Thank you for your patience helping me. I admit, I'm using vi in the terminal for all of my edits.

Dusch4593 commented:
> Thank you for your patience helping me. I admit, I'm using vi in the terminal for all of my edits.

I'm always happy to help 😄 and kudos for working with the repo through the `vi` editor!

Dusch4593 commented:
@beverand Would you be open to trying to format the entry on your side with the following command when you get a chance?

```
yarn format content/python/concepts/built-in-functions/terms/super/super.md content/python/concepts/inheritance/inheritance.md
```

Dusch4593 commented:
> @beverand Would you be open to trying to format the entry on your side with the following command when you get a chance?
> 
> ```
> yarn format content/python/concepts/built-in-functions/terms/super/super.md content/python/concepts/inheritance/inheritance.md
> ```

And then `add`, `commit`, and `push` the changes like you normally would?

beverand commented:
I think there are things I have to install to make that work

On Fri, Sep 16, 2022 at 4:21 PM Brandon Dusch ***@***.***>
wrote:

> @beverand <https://github.com/beverand> Would you be open to trying to
> format the entry on your side with the following command when you get a
> chance?
>
> yarn format content/python/concepts/built-in-functions/terms/super/super.md content/python/concepts/inheritance/inheritance.md
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Codecademy/docs/pull/1008#issuecomment-1249931669>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEZUG6XITOB5UZ2QZW3QWLTV6T6F7ANCNFSM6AAAAAAQOCNCAQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

beverand commented:
I am getting errors. Something with prettier or something with Arborist.

On Fri, Sep 16, 2022 at 4:31 PM Beverly Andrews ***@***.***>
wrote:

> I think there are things I have to install to make that work
>
> On Fri, Sep 16, 2022 at 4:21 PM Brandon Dusch ***@***.***>
> wrote:
>
>> @beverand <https://github.com/beverand> Would you be open to trying to
>> format the entry on your side with the following command when you get a
>> chance?
>>
>> yarn format content/python/concepts/built-in-functions/terms/super/super.md content/python/concepts/inheritance/inheritance.md
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/Codecademy/docs/pull/1008#issuecomment-1249931669>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AEZUG6XITOB5UZ2QZW3QWLTV6T6F7ANCNFSM6AAAAAAQOCNCAQ>
>> .
>> You are receiving this because you were mentioned.Message ID:
>> ***@***.***>
>>
>

Dusch4593 commented:
> I am getting errors. Something with prettier or something with Arborist. On Fri, Sep 16, 2022 at 4:31 PM Beverly Andrews ***@***.***> wrote:
> […](#)
> I think there are things I have to install to make that work On Fri, Sep 16, 2022 at 4:21 PM Brandon Dusch ***@***.***> wrote: > @beverand <https://github.com/beverand> Would you be open to trying to > format the entry on your side with the following command when you get a > chance? > > yarn format content/python/concepts/built-in-functions/terms/super/super.md content/python/concepts/inheritance/inheritance.md > > — > Reply to this email directly, view it on GitHub > <[#1008 (comment)](https://github.com/Codecademy/docs/pull/1008#issuecomment-1249931669)>, > or unsubscribe > <https://github.com/notifications/unsubscribe-auth/AEZUG6XITOB5UZ2QZW3QWLTV6T6F7ANCNFSM6AAAAAAQOCNCAQ> > . > You are receiving this because you were mentioned.Message ID: > ***@***.***> >

Any chance you could copy/paste the exact error message here?

beverand commented:
I've tried running yarn format super.md and also npm run format super.
It's telling me jest: not found

On Fri, Sep 16, 2022 at 4:42 PM Beverly Andrews ***@***.***>
wrote:

> I am getting errors. Something with prettier or something with Arborist.
>
> On Fri, Sep 16, 2022 at 4:31 PM Beverly Andrews ***@***.***>
> wrote:
>
>> I think there are things I have to install to make that work
>>
>> On Fri, Sep 16, 2022 at 4:21 PM Brandon Dusch ***@***.***>
>> wrote:
>>
>>> @beverand <https://github.com/beverand> Would you be open to trying to
>>> format the entry on your side with the following command when you get a
>>> chance?
>>>
>>> yarn format content/python/concepts/built-in-functions/terms/super/super.md content/python/concepts/inheritance/inheritance.md
>>>
>>> —
>>> Reply to this email directly, view it on GitHub
>>> <https://github.com/Codecademy/docs/pull/1008#issuecomment-1249931669>,
>>> or unsubscribe
>>> <https://github.com/notifications/unsubscribe-auth/AEZUG6XITOB5UZ2QZW3QWLTV6T6F7ANCNFSM6AAAAAAQOCNCAQ>
>>> .
>>> You are receiving this because you were mentioned.Message ID:
>>> ***@***.***>
>>>
>>

beverand commented:
Will this do?

[image: image.png]


On Fri, Sep 16, 2022 at 4:43 PM Brandon Dusch ***@***.***>
wrote:

> I am getting errors. Something with prettier or something with Arborist.
> On Fri, Sep 16, 2022 at 4:31 PM Beverly Andrews *@*.
>
> *> wrote: … <#m_5599583300834726542_> I think there are things I have to
> install to make that work On Fri, Sep 16, 2022 at 4:21 PM Brandon Dusch @.*>
> wrote: > @beverand <https://github.com/beverand>
> https://github.com/beverand Would you be open to trying to > format the
> entry on your side with the following command when you get a > chance? > >
> yarn format content/python/concepts/built-in-functions/terms/super/super.md
> content/python/concepts/inheritance/inheritance.md > > — > Reply to this
> email directly, view it on GitHub > <#1008 (comment)
> <https://github.com/Codecademy/docs/pull/1008#issuecomment-1249931669>>,
> > or unsubscribe >
> https://github.com/notifications/unsubscribe-auth/AEZUG6XITOB5UZ2QZW3QWLTV6T6F7ANCNFSM6AAAAAAQOCNCAQ
> > . > You are receiving this because you were mentioned.Message ID: > *@*.***>
> >
>
> Any chance you could copy/paste the exact error message here?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Codecademy/docs/pull/1008#issuecomment-1249939127>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEZUG6WPRDM5UXOPQKPMLZTV6UA2BANCNFSM6AAAAAAQOCNCAQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

beverand commented:
I'm doing npm install --force
Maybe that will fix it...

On Fri, Sep 16, 2022 at 4:49 PM Beverly Andrews ***@***.***>
wrote:

> Will this do?
>
> [image: image.png]
>
>
> On Fri, Sep 16, 2022 at 4:43 PM Brandon Dusch ***@***.***>
> wrote:
>
>> I am getting errors. Something with prettier or something with Arborist.
>> On Fri, Sep 16, 2022 at 4:31 PM Beverly Andrews *@*.
>>
>> *> wrote: … <#m_-356230180408197519_m_5599583300834726542_> I think there
>> are things I have to install to make that work On Fri, Sep 16, 2022 at 4:21
>> PM Brandon Dusch @.*> wrote: > @beverand <https://github.com/beverand>
>> https://github.com/beverand Would you be open to trying to > format the
>> entry on your side with the following command when you get a > chance? > >
>> yarn format content/python/concepts/built-in-functions/terms/super/super.md
>> content/python/concepts/inheritance/inheritance.md > > — > Reply to this
>> email directly, view it on GitHub > <#1008 (comment)
>> <https://github.com/Codecademy/docs/pull/1008#issuecomment-1249931669>>,
>> > or unsubscribe >
>> https://github.com/notifications/unsubscribe-auth/AEZUG6XITOB5UZ2QZW3QWLTV6T6F7ANCNFSM6AAAAAAQOCNCAQ
>> > . > You are receiving this because you were mentioned.Message ID: > *@*.***>
>> >
>>
>> Any chance you could copy/paste the exact error message here?
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/Codecademy/docs/pull/1008#issuecomment-1249939127>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AEZUG6WPRDM5UXOPQKPMLZTV6UA2BANCNFSM6AAAAAAQOCNCAQ>
>> .
>> You are receiving this because you were mentioned.Message ID:
>> ***@***.***>
>>
>

beverand commented:
It looks like it worked.

On Fri, Sep 16, 2022 at 4:53 PM Beverly Andrews ***@***.***>
wrote:

> I'm doing npm install --force
> Maybe that will fix it...
>
> On Fri, Sep 16, 2022 at 4:49 PM Beverly Andrews ***@***.***>
> wrote:
>
>> Will this do?
>>
>> [image: image.png]
>>
>>
>> On Fri, Sep 16, 2022 at 4:43 PM Brandon Dusch ***@***.***>
>> wrote:
>>
>>> I am getting errors. Something with prettier or something with Arborist.
>>> On Fri, Sep 16, 2022 at 4:31 PM Beverly Andrews *@*.
>>>
>>> *> wrote: …
>>> <#m_-801615125820641623_m_-356230180408197519_m_5599583300834726542_> I
>>> think there are things I have to install to make that work On Fri, Sep 16,
>>> 2022 at 4:21 PM Brandon Dusch @.*> wrote: > @beverand
>>> <https://github.com/beverand> https://github.com/beverand Would you be
>>> open to trying to > format the entry on your side with the following
>>> command when you get a > chance? > > yarn format
>>> content/python/concepts/built-in-functions/terms/super/super.md
>>> content/python/concepts/inheritance/inheritance.md > > — > Reply to this
>>> email directly, view it on GitHub > <#1008 (comment)
>>> <https://github.com/Codecademy/docs/pull/1008#issuecomment-1249931669>>,
>>> > or unsubscribe >
>>> https://github.com/notifications/unsubscribe-auth/AEZUG6XITOB5UZ2QZW3QWLTV6T6F7ANCNFSM6AAAAAAQOCNCAQ
>>> > . > You are receiving this because you were mentioned.Message ID: >
>>> *@*.***> >
>>>
>>> Any chance you could copy/paste the exact error message here?
>>>
>>> —
>>> Reply to this email directly, view it on GitHub
>>> <https://github.com/Codecademy/docs/pull/1008#issuecomment-1249939127>,
>>> or unsubscribe
>>> <https://github.com/notifications/unsubscribe-auth/AEZUG6WPRDM5UXOPQKPMLZTV6UA2BANCNFSM6AAAAAAQOCNCAQ>
>>> .
>>> You are receiving this because you were mentioned.Message ID:
>>> ***@***.***>
>>>
>>

Dusch4593 commented:
@beverand Yup that did it, thanks!

beverand commented:
Closes #586

beverand commented:
I'm not sure what to do now.

Dusch4593 reviewed (approved):
Hey @beverand, sorry for the delay! These changes look good to me from my end, so let's have the next reviewer take a look at this. 😄

beverand commented:
Yay!

On Tue, Sep 20, 2022 at 9:28 AM Brandon Dusch ***@***.***>
wrote:

> ***@***.**** approved this pull request.
>
> Hey @beverand <https://github.com/beverand>, sorry for the delay! These
> changes look good to me from my end, so let's have the next reviewer take a
> look at this. 😄
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Codecademy/docs/pull/1008#pullrequestreview-1114143394>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEZUG6QLBCAY7YRTFYOT5ETV7HQ33ANCNFSM6AAAAAAQOCNCAQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

KTom101 commented:
@beverand Taking a look at this now. I will leave my review when I'm done.

KTom101 reviewed (changes requested):
@beverand I've added a few changes to this. Also, please remove the yarn.lock file. Let me know if there are any questions.

beverand commented on content/python/concepts/built-in-functions/terms/super/super.md:
````diff
@@ -0,0 +1,53 @@
+super().function()
+super(type)
+```
+
````
Updated "function" to "method" and put a line to indicate what the things are. Not sure if it is a good explanation, trying to stay focused on super without bleeding into other dense topics.

beverand replied:
I've moved the text to be below the pseudo section

beverand commented:
I removed the lock file, but now the actions are failing? Please advise.

Dusch4593 commented:
Hey @beverand my team and I were able to rectify the failing checks over a few commits by doing the following:

- Replaced the diff'd state of the `yarn.lock` file with it's current state under `Codecademy/docs:main`.
- Caught your PR up with recent changes in `Codecademy/docs:main` 😄

@KTom101 How are things looking with this? Everything good to go for merge?

KTom101 commented on content/python/concepts/built-in-functions/terms/super/super.md:
````diff
@@ -0,0 +1,55 @@
+
+```pseudo
+super().method()
+super(type)
````
Can we show how the `super(type)` is used in its own example?

KTom101 commented on content/python/concepts/built-in-functions/terms/super/super.md:
```diff
@@ -0,0 +1,55 @@
+## `super()`
+
+`super()` returns a temporary object of the superclass, allowing the superclass’s methods to be called. This comes in handy when using multiple inheritance to derive a class from more than one base class.
+
```
Add ## Example subheading. Give an example and brief explanation showing how `super(type)` syntax works.

beverand replied:
I made the update to super(type) by updating it to super(type, object) and creating an additional codebyte example for it.

KTom101 reviewed (commented):
@beverand I've added a change to the update. Also, since the `super(type)` syntax is mentioned. An example section should be added before the codebyte example showing how it is used.

beverand commented:
@Dusch4593 thank you so much for fixing the lock file thing.

Dusch4593 commented on content/python/concepts/built-in-functions/terms/super/super.md:
````diff
@@ -0,0 +1,74 @@
+```
+
+`.method()` is the accessed method of the parent class. `type` specifies the type object of the parent class and object is an instance or subtype of the `type`.
+
````
Can we place a second \```pseudo snippet here that looks like this:

```pseudo
class A():
  method(arg):
    # Method code starts here

class B(A):
  method(arg):
    super().method(arg)
    # Method code starts here
```

beverand replied:
Updated.

Dusch4593 reviewed (changes requested):
Hey @beverand! I found a few last areas for suggestions and comments on my end before we merge. After you've implemented this last batch of feedback and @KTom101 and I approve the changes, we should be good for merge. 😄

Btw, I just wanna give you kudos for the work you've put into this PR this week. This is an incredibly important and oft-used Python function that has a lot packed into it. Major props! 🚀

KTom101 reviewed (commented):
@beverand Added a few more changes, mainly concerning the second codebyte.  We only want one codebyte on the page and it should be at the end. I've added changes to reflect that.

beverand commented on content/python/concepts/inheritance/inheritance.md:
```diff
@@ -13,7 +13,7 @@ CatalogContent:
   - 'paths/computer-science'
```
Added.

beverand commented on content/python/concepts/built-in-functions/terms/super/super.md:
````diff
@@ -0,0 +1,74 @@
+
+```pseudo
+super().method()
+super(type, object)
````
Fixed super().method() to super()

KTom101 reviewed (approved):
@beverand @Dusch4593 The new changes look great! 👍

KTom101 commented:
Live links are up!
`super()` : https://www.codecademy.com/resources/docs/python/built-in-functions/super
Inheritance : https://www.codecademy.com/resources/docs/python/inheritance
